feat(#377): list failing tests

renancaraujo opened the pull request:
## Description

Adds a listing of failing tests after a test run as proposed on #377


![Screenshot 2022-05-13 at 17 52 10](https://user-images.githubusercontent.com/6718144/168332078-d7f5f2ff-d807-4050-828f-032f7eff06ea.png)





## Type of Change

<!--- Put an `x` in all the boxes that apply: -->

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

renancaraujo commented:
Changed it to have an output including line and column, as well as the path of to the suite.
The only problem is that the suite is always identified as `./test/.test_runner.dart` when optimization is on (default).

Without optimization, the output is: 
```
Failing Tests:
 - <path to cwd>/test/src/widgets/the_real_test_file.dart:30:5
```

With optimization, the output is: 
```
Failing Tests:
 - <path to cwd>/test/.test_runner.dart:30:5
```

renancaraujo commented on lib/src/cli/flutter_cli.dart:
```diff
@@ -366,6 +366,22 @@ Future<int> _flutterTest({
+              return previousValue;
+            },
+          );
+          stdout(lines.toString());
```
Changed to output the suite paths. I see no problem on wrapping them. What do you think?

renancaraujo replied:
Also changed to output via `stderr`. I was going to wrap everything on red but since the `Logger.error` does that for us, it was not necessary.

felangel commented on lib/src/cli/flutter_cli.dart:
```diff
@@ -270,21 +271,22 @@ Future<int> _flutterTest({
   List<String>? arguments,
   required void Function(String) stdout,
   required void Function(String) stderr,
+  required bool optimizePerformance,
```
Do we need this parameter? I didn't see it being used anywhere.

renancaraujo commented:
Update: Changed to retrieve the test location from the stack trace. In this way, the right path is recovered regardless of the optimization being true of false .
Also added a prefix to show if the test failed  or resulted in an error:

```
Failing Tests:
 - [FAILED] test/example_test.dart 4:5
 - [ERROR] test/example_test.dart 5:5
```

wolfenrain commented on test/src/cli/flutter_cli_test.dart:
```diff
@@ -381,6 +381,11 @@ void main() {
         ).called(1);
+        verify(
+          () => logger.err(
+            any(that: contains('- [FAILED] test/example_test.dart 4:5')),
```
The output here, can we somehow turn that into:
```
[FAILED] test/example_test.dart:4:5
```

Most IDEs know that this means a certain file from the current root at given column and line. This would allow developers to immediately jump to the correct line.

renancaraujo replied:
Sure 👍

renancaraujo replied:
Done

felangel commented:
<img width="709" alt="Screen Shot 2022-05-19 at 10 15 08 AM" src="https://user-images.githubusercontent.com/8855632/169331920-a76bb4c0-110b-4120-ab00-3448e0a5748a.png">

felangel reviewed (approved):
LGTM 💯
